docs/4.0/ssh_gsuite.md

```diff
@@ -0,0 +1,160 @@
+spec:
+  redirect_url: https://localhost:3080/v1/webapi/oidc/callback
+  client_id: exampleclientid11234.apps.googleusercontent.com
+  client_secret: examplesecret
+  issuer_url: https://accounts.google.com
+  scope: ['https://www.googleapis.com/auth/admin.directory.group.readonly', 'openid', 'email']  
+  claims_to_roles:
+    - {claim: "groups", value: "admin@example.com", roles: ["admin"]}
+```
+
+!!! important 
+    The groups will be fetched only if admins include special auth scope https://www.googleapis.com/auth/admin.directory.group.readonly in the scopes of the connector as shown in the example above.
+
+
+Create the connector using `tctl` tool:
+
+```bsh
+$ tctl create gsuite-connector.yaml
+```
+
+## Create Teleport Roles
+
+We are going to create 2 roles:
+-  privileged role admin who is able to login as root and is capable of administrating 
+the cluster and non-privileged dev.
+- non-privileged dev
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: admin
+spec:
+  options:
+    max_session_ttl: 24h
+  allow:
+    logins: [root]
+    node_labels:
+      "*": "*"
+    rules:
+      - resources: ["*"]
+        verbs: ["*"]
+```
+
+Devs are only allowed to login to nodes labelled with `access: relaxed`
+Teleport label. Developers can log in as either `ubuntu` or a username that
+arrives in their assertions. Developers also do not have any rules needed to
+obtain admin access to Teleport.
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: dev
+spec:
+  options:
+    max_session_ttl: 24h
+  allow:
+    logins: [ "{{external.username}}", ubuntu ]
+    node_labels:
+      access: relaxed
+```
+
+**Notice:** Replace `ubuntu` with linux login available on your servers!
+
+```bsh
+$ tctl create admin.yaml
+$ tctl create dev.yaml
+```
+
+## Testing
+![Login with Gsuite](img/gsuite/gsuite-6-loginwithgsuite.png)
+
+
+The Web UI will now contain a new button: "Login with GSuite". The CLI is
+the same as before:
+
+```bsh
+$ tsh --proxy=proxy.example.com login
+```
+
+This command will print the SSO login URL (and will try to open it
+automatically in a browser).
+
+!!! tip "Tip":
+    Teleport can use multiple OIDC connectors. In this case a connector name
+    can be passed via `tsh login --auth=connector_name`
+
+
+## Troubleshooting
+
+If you get "access denied" errors the number one place to check is the audit
+log on the Teleport auth server. It is located in `/var/lib/teleport/log` by
+default and it will contain the detailed reason why a user's login was denied.
+
+Example of a user being denied due as the role `clusteradmin` wasn't setup. 
+```json
+{"code":"T1001W","error":"role clusteradmin is not found","event":"user.login","method":"oidc","success":false,"time":"2019-06-15T19:38:07Z","uid":"cd9e45d0-b68c-43c3-87cf-73c4e0ec37e9"}
+```
+
+
+Some errors (like filesystem permissions or misconfigured network) can be
+diagnosed using Teleport's `stderr` log, which is usually available via:
+
+```bsh
+$ sudo journalctl -f teleport
```

Comment:
I don't think that this command will actually work as intended without the `-u`. Should be `sudo journalctl -u teleport -f` or whatever.

Reply:
ha, I checked and only -fu works, nor -uf. Going to switch it back.